CONDUCT.md

```diff
@@ -1,12 +1,94 @@
+
+The second goal is to provide a mechanism for resolving conflicts in the community when they arise.
+
+The third goal of the Code of Conduct is to make our community welcoming to people from different backgrounds. Diversity is critical to the project; for Gonum to be successful, it needs contributors and users from all backgrounds. (See [Go, Open Source, Community.](https://blog.golang.org/open-source))
+
+We believe that healthy debate and disagreement are essential to a healthy project and community. However, it is never ok to be disrespectful. We value diverse opinions, but we value respectful behavior more.
+
+## Numerical Gopher values
+
+These are the values to which people in the Gonum community (“Numerical Gophers”) should aspire.
 
 - Be friendly and welcoming
 - Be patient
+	- Remember that people have varying communication styles and that not everyone is using their native language. (Meaning and tone can be lost in translation.) 
 - Be thoughtful
+	- Productive communication requires effort. Think about how your words will be interpreted.
+	- Remember that sometimes it is best to refrain entirely from commenting. 
 - Be respectful
+	- In particular, respect differences of opinion. 
 - Be charitable
-- Avoid destructive behavior
+	- Interpret the arguments of others in good faith, do not seek to disagree.
+	- When we do disagree, try to understand why. 
+- Avoid destructive behavior:
+	- Derailing: stay on topic; if you want to talk about something else, start a new conversation.
+	- Unconstructive criticism: don't merely decry the current state of affairs; offer—or at least solicit—suggestions as to how things may be improved.
+	- Snarking (pithy, unproductive, sniping comments)
+	- Discussing potentially offensive or sensitive issues; this all too often leads to unnecessary conflict.
+	- Microaggressions: brief and commonplace verbal, behavioral and environmental indignities that communicate hostile, derogatory or negative slights and insults to a person or group. 
+
+People are complicated. You should expect to be misunderstood and to misunderstand others; when this inevitably occurs, resist the urge to be defensive or assign blame. Try not to take offense where no offense was intended. Give people the benefit of the doubt. Even if the intent was to provoke, do not rise to it. It is the responsibility of *all parties* to de-escalate conflict when it arises.
+
+## Code of Conduct
+
+### Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
```

Comment:
Why is this list slightly different than the list at the top of the file? Could we use this one (which is more complete) in both places?